code/game/objects/items/weapons/bio_chips/bio_chip_death_alarm.dm

```diff
@@ -26,8 +28,12 @@
 			a.autosay("[mobname] has died-zzzzt in-in-in...", "[mobname]'s Death Alarm")
 			qdel(src)
 		if("emp")
+			if(emp_spam_lock)
+				return
 			var/name = prob(50) ? t.name : pick(SSmapping.teleportlocs)
 			a.autosay("[mobname] has died in [name]!", "[mobname]'s Death Alarm")
+			emp_spam_lock = TRUE
+			addtimer(VARSET_CALLBACK(src, emp_spam_lock, FALSE), 0.1 SECONDS, TIMER_UNIQUE | TIMER_OVERRIDE) // shamelessly stolen from pulse demon code
```

Comment:
use the cooldowns from tg_cooldowns.dm instead (COOLDOWN_FINISHED, COOLDOWN_START, COOLDOWN_DECLARE)